styx-api-service/src/main/java/com/spotify/styx/api/StatusResource.java

```diff
@@ -139,6 +144,29 @@ private static String arg(String name, RequestContext rc) {
     return Response.forPayload(RunStateDataPayload.create(runStates));
   }
 
+  private Map<WorkflowInstance, RunState> getActiveStates(String componentsStr) throws IOException {
+    final List<String> components = Arrays.asList(componentsStr.split(","));
+    final ImmutableMap.Builder<WorkflowInstance, RunState> mapBuilder = ImmutableMap.builder();
+
+    final List<Optional<IOException>> exceptions = components.parallelStream().map( componentId -> {
```

Comment:
I think we have some problems here:
* `parallelStream` for IO blocking operation is not good because it uses common FJP, [this](https://github.com/spotify/styx/blob/master/styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java#L188) is a better way
* The map is not thread safe and we have multiple threads adding things to it